src/useReactions.test.tsx

```diff
@@ -0,0 +1,242 @@
+  public memberships = Object.entries(membership).map(([eventId, sender]) => ({
+    sender,
+    eventId,
+    createdTs: (): Date => new Date(),
+  }));
+
+  public constructor(public readonly room: MockRoom) {
+    super();
+  }
+
+  public testRemoveMember(userId: string): void {
+    this.memberships = this.memberships.filter((u) => u.sender !== userId);
+    this.emit(MatrixRTCSessionEvent.MembershipsChanged);
+  }
+
+  public testAddMember(sender: string): void {
+    this.memberships.push({
+      sender,
+      eventId: `!fake-${randomUUID()}:event`,
+      createdTs: (): Date => new Date(),
+    });
+    this.emit(MatrixRTCSessionEvent.MembershipsChanged);
+  }
+}
+
+function createReaction(parentMemberEvent: string): MatrixEvent {
+  return new MatrixEvent({
+    sender: membership[parentMemberEvent],
+    type: EventType.Reaction,
+    origin_server_ts: new Date().getTime(),
+    content: {
+      "m.relates_to": {
+        key: "🖐️",
+        event_id: parentMemberEvent,
+      },
+    },
+    event_id: randomUUID(),
+  });
+}
+
+function createRedaction(sender: string, reactionEventId: string): MatrixEvent {
+  return new MatrixEvent({
+    sender,
+    type: EventType.RoomRedaction,
+    origin_server_ts: new Date().getTime(),
+    redacts: reactionEventId,
+    content: {},
+    event_id: randomUUID(),
+  });
+}
+
+export class MockRoom extends EventEmitter {
+  public constructor(private readonly existingRelations: MatrixEvent[] = []) {
+    super();
+  }
+
+  public get client(): MatrixClient {
+    return {
+      getUserId: (): string => memberUserIdAlice,
+    } as unknown as MatrixClient;
+  }
+
+  public get relations(): Room["relations"] {
+    return {
+      getChildEventsForEvent: (membershipEventId: string) => ({
+        getRelations: (): MatrixEvent[] => {
+          const sender = membership[membershipEventId];
+          return this.existingRelations.filter((r) => r.getSender() === sender);
+        },
+      }),
+    } as unknown as Room["relations"];
+  }
+
+  public testSendReaction(parentMemberEvent: string): string {
+    const evt = createReaction(parentMemberEvent);
+    this.emit(RoomEvent.Timeline, evt, this, undefined, false, {
+      timeline: new EventTimeline(new EventTimelineSet(undefined)),
+    });
+    return evt.getId()!;
+  }
+}
+
+describe("useReactions", () => {
+  test("starts with an empty list", () => {
+    const rtcSession = new MockRTCSession(new MockRoom());
+    const { queryByRole } = render(
+      <TestComponentWrapper rtcSession={rtcSession} />,
+    );
+    expect(queryByRole("list")?.children).to.have.lengthOf(0);
+  });
+  test("handles own raised hand", () => {
+    const room = new MockRoom();
+    const rtcSession = new MockRTCSession(room);
+    const { queryByText, rerender } = render(
+      <TestComponentWrapper rtcSession={rtcSession} />,
+    );
+    room.testSendReaction(memberEventAlice);
+    rerender(<TestComponentWrapper rtcSession={rtcSession} />);
```

Comment:
Hm, in React Testing Library I believe the preferred pattern is to let the component update on its own. Calling `rerender` with the same props is effectively forcing a render and calling into question whether the reactivity of `useReactions` is broken. You may need to wrap the update in `act` for it to take, though.

```suggestion
    const { queryByText } = render(
      <TestComponentWrapper rtcSession={rtcSession} />,
    );
    await act(async () => room.testSendReaction(memberEventAlice));
```

Reply:
Yep, using `act` works :)